### PR TITLE
Add SevenNet to MatCalc

### PR DESCRIPTION
**Pull request by AntObi**
## Summary

Major changes:

- Add SevenNet support for `get_universal_calculator()` 
- Add the SevenNet package (`sevenn`) to optional pyproject.toml dependencies
- Add SevenNet MLIP to the `examples/Calculating MLIP properties.ipynb` notebook
- Import `ExpCellFilter` from `ase.filters` instead of `ase.constraints` in `relaxation.py` 
- Update plots in `examples/Calculating MLIP properties.ipynb`

## Checklist

- [x] Google format doc strings added. Check with `ruff`.
- [x] Type annotations included. Check with `mypy`.
- [x] Tests added for new features/fixes.
- [x] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,7 +39,7 @@ dependencies = ["ase", "joblib", "phonopy", "pymatgen"]
 version = "0.0.4"
 
 [project.optional-dependencies]
-models = ["chgnet", "mace-torch", "matgl>=0.9.0"]
+models = ["chgnet", "mace-torch", "matgl>=0.9.0", "sevenn"]
```
Comment (janosh) [inline pyproject.toml:42]: let's add a minimum version pin to all models pointing at the latest pypi version in all cases. older versions have known bugs in many cases and should not be used

File: matcalc/relaxation.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import pickle
 from typing import TYPE_CHECKING
 
-from ase.constraints import ExpCellFilter
+from ase.filters import ExpCellFilter
```
Comment (janosh) [inline matcalc/relaxation.py]: let's change this to the new released `FrechetCellFilter` while we're at it which has [improved convergence behavior](https://gitlab.com/ase/ase/-/merge_requests/3024)

Comment (janosh) [reply matcalc/relaxation.py]: also, could you recreate the figures since the relaxation times might decrease a bit?

File: pyproject.toml [hunk 3]
```diff
@@ -39,7 +39,7 @@ dependencies = ["ase", "joblib", "phonopy", "pymatgen"]
 version = "0.0.4"
```
Comment (janosh) [inline pyproject.toml:39]: let's also set a down-pin to the latest `ase`

**Review by janosh — CHANGES_REQUESTED** [review]
nice work! thanks @AntObi. just some minor suggestions

File: pyproject.toml (hunk 1 above)
Comment (AntObi) [reply pyproject.toml:42]: Will do

File: matcalc/relaxation.py (hunk 2 above)
Comment (AntObi) [reply matcalc/relaxation.py]: Can do

File: pyproject.toml (hunk 3 above)
Comment (AntObi) [reply pyproject.toml:39]: Will do

Comment (AntObi) [reply pyproject.toml:39]: Pinned to "ase>=3.23.0"

File: pyproject.toml (hunk 1 above)
Comment (AntObi) [reply pyproject.toml:42]: Pinned

File: matcalc/relaxation.py (hunk 2 above)
Comment (AntObi) [reply matcalc/relaxation.py]: @janosh Happy to implement this. Implementing this however, will require me to rewrite some of the tests as some of the results will be different with the `FrechetCellFilter`.

For `PropCalc` classes which have the option `relax_structure`, I think an option to pass kwargs to the `RelaxCalc`, if the user desires. Happy to also implement this too!

File: pyproject.toml (hunk 3 above)
Comment (janosh) [reply pyproject.toml:39]: maybe you forgot to push?

File: matcalc/relaxation.py (hunk 2 above)
Comment (janosh) [reply matcalc/relaxation.py]: both would be great! 🙏

Comment (AntObi) [reply matcalc/relaxation.py]: Added the relax kwargs to the other `PropCalc` classes, and have tested this using the tests by passing `{"cell_filter", ExpCellFilter}` to get the same results prior to making `FrechetCellFilter` the default. I've also added some tests in `test_relaxation.py` to test the outputs of the new default `FrechetCellFilter`.

**Review by janosh — APPROVED** [review]
thanks a lot @AntObi for adding SevenNet! very useful 👍

**Pull request merged**